src/main/java/address/browser/AddressBookBrowser.java

```diff
@@ -0,0 +1,273 @@
+package address.browser;
+
+import address.model.ModelPerson;
+import address.model.Person;
+
+import address.util.Range;
+import com.teamdev.jxbrowser.chromium.Browser;
+import com.teamdev.jxbrowser.chromium.EditorCommand;
+import com.teamdev.jxbrowser.chromium.dom.By;
+import com.teamdev.jxbrowser.chromium.dom.DOMElement;
+import com.teamdev.jxbrowser.chromium.dom.events.DOMEventType;
+import com.teamdev.jxbrowser.chromium.events.FinishLoadingEvent;
+import com.teamdev.jxbrowser.chromium.events.LoadAdapter;
+import com.teamdev.jxbrowser.chromium.javafx.BrowserView;
+import javafx.collections.ObservableList;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TabPane;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A browser for displaying AddressBook contacts' profile page.
+ */
+public class AddressBookBrowser{
+
+    private static final String INSTRUCTION_PAGE_HTML_CODE =
+            "<html><body><h3>To view contact's web page, click on the contact on the left.</h3></body></html>";
+
+    private ArrayList<BrowserTab> pages;
+    private ObservableList<ModelPerson> filteredModelPersons;
+    private TabPane addressBookBrowserView;
+
+    public AddressBookBrowser(int noOfTabs, ObservableList<ModelPerson> filteredModelPersons) {
+        addressBookBrowserView = new TabPane();
+        this.filteredModelPersons = filteredModelPersons;
+        this.pages = new ArrayList<BrowserTab>();
+        for (int i = 0; i < noOfTabs; i++) {
+            pages.add(new BrowserTab());
+            Tab tab = createBrowserTabView(pages.get(i));
+            addressBookBrowserView.getTabs().add(tab);
+        }
+        hideTabs();
+        init();
+    }
+
+    private void hideTabs() {
+        addressBookBrowserView.setVisible(false);
+        addressBookBrowserView.setTabMaxHeight(0);
+    }
+
+    /**
+     * Initialises browser
+     *
+     * This includes:
+     - Loads instruction page on browser
+     */
+    public void init() {
+        pages.get(0).loadHTML(INSTRUCTION_PAGE_HTML_CODE);
```

Comment:
If we are not initialising everything here (`hideTabs` sounds like part of `init`, but it's separated), then we might not need `init()`. This method can be renamed to `loadInstructionToTab` or something similar.